### ISSUE TITLE
Related pickers crashes the app

### ISSUE THREAD

**Issue by clamiax** [react-native-community/react-native-picker#14]
It seems impossible to bind two pickers together (one changing depending on the value of the other) because in 99% of cases the app crashes. It's a bug somewhere in the RN code which I was not able to workaround. The snippet below should make everything clears.

For now, I was forced to replace the pickers with custom modals.

React Native version:
System:
    OS: Linux 4.18 Ubuntu 18.10 (Cosmic Cuttlefish)
    CPU: (4) x64 Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
    Memory: 1.91 GB / 3.73 GB
    Shell: Unknown - /bin/mksh
  Binaries:
    Node: 8.11.4 - /usr/bin/node
    npm: 6.10.3 - /usr/local/bin/npm
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.4 => 0.60.4
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-screens: 1.0.0-alpha.23

## Steps To Reproduce

1. Run the code below
2. Select `Bar > BarSub02` then go back to `Foo`

I expect to find `Foo > FooSub01` selected. I get a FATAL EXCEPTION instead.

Following code example is a reproducible test case:
```
import React, {Component} from "react";
import {
        Picker,
        View
} from "react-native";

export default class PickerTest extends Component {
        constructor(props) {
                super(props);
                this.state = {
                        index_group: 0,
                        index_subgroup: 0,
                        items: [
                                {name:"Foo", subs: [
                                        {ClsSubGrName:"FooSub01"},
                                ]},
                                {name:"Bar", subs: [
                                        {ClsSubGrName:"BarSub01"},
                                        {ClsSubGrName:"BarSub02"}
                                ]}
                        ]
                };
        }

        render() {
                return (
                <View style={{flex:1}}>
                        <Picker
                                selectedValue={this.state.index_group}
                                onValueChange={(v) => this.setState({index_group:v,index_subgroup:0})}>
                                {this.state.items.map((item, i) => (
                                        <Picker.Item key={i} value={i} label={item.name} />
                                ))}
                        </Picker>

                        <Picker
                                selectedValue={this.state.index_subgroup}
                                onValueChange={(v) => this.setState({index_subgroup:v})}>
                                {this.state.items[this.state.index_group].subs.map((item, i) => (
                                        <Picker.Item key={i} value={i} label={item.ClsSubGrName} />
                                ))}
                        </Picker>
                </View>
                )
        }
}
```

Notice that the issue does not occur if current item is the first in list or if you switch between groups with more than one item (e.g. if you add `FooSub02` under `Foo` the issue should not happen anymore).

Here's an excerpt from my logs:

```
08-09 13:11:27.589  3658  3658 E unknown:ReactNative: Exception in native call
08-09 13:11:27.589  3658  3658 E unknown:ReactNative: java.lang.ArrayIndexOutOfBoundsException: length=1; index=3
08-09 13:11:27.589  3658  3658 E unknown:ReactNative:   at java.util.Arrays$ArrayList.get(Arrays.java:3854)
[ ...OMISSIS... ]
08-09 13:11:27.594  3658  3658 D AndroidRuntime: Shutting down VM
08-09 13:11:27.595  3658  3658 E AndroidRuntime: FATAL EXCEPTION: main
```

**Comment by jainkuniya** [comment]
hi @clamiax I tried to reproduce, but can't reproduce
```
 "@react-native-community/picker": "^1.0.0",
 "react": "16.8.6",
 "react-native": "0.60.4"
```

My steps were:
> Select Bar > BarSub02 then go back to Foo


![ezgif com-video-to-gif (3)](https://user-images.githubusercontent.com/18511177/62819999-1d646100-bb7b-11e9-8a97-d1fa9ffc051e.gif)

**Comment by clamiax** [comment]
@jainkuniya Thanks for taking a look.

I'm still experiencing the issue, I also tried on a real device with the exact same effect. Here's the full log:

```
08-12 10:42:00.382 20271 20271 E unknown:ReactNative: Exception in native call
08-12 10:42:00.382 20271 20271 E unknown:ReactNative: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at java.util.Arrays$ArrayList.get(Arrays.java:3854)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.widget.ArrayAdapter.getItem(ArrayAdapter.java:349)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.views.picker.ReactPickerManager$ReactPickerAdapter.getView(ReactPickerManager.java:116)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.views.picker.ReactPickerManager$ReactPickerAdapter.getView(ReactPickerManager.java:107)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.widget.Spinner.makeView(Spinner.java:713)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.widget.Spinner.layout(Spinner.java:661)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.widget.AbsSpinner.setSelectionInt(AbsSpinner.java:301)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.widget.AbsSpinner.setSelection(AbsSpinner.java:278)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.views.picker.ReactPicker.commitStagedData(ReactPicker.java:142)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.views.picker.ReactPickerManager.onAfterUpdateTransaction(ReactPickerManager.java:80)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.views.picker.ReactPickerManager.onAfterUpdateTransaction(ReactPickerManager.java:36)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:139)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:844)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:952)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.UIViewOperationQueue.access$2200(UIViewOperationQueue.java:44)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1012)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:172)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:84)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.view.Choreographer.doCallbacks(Choreographer.java:768)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.view.Choreographer.doFrame(Choreographer.java:696)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:955)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.os.Handler.handleCallback(Handler.java:846)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.os.Handler.dispatchMessage(Handler.java:106)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.os.Looper.loop(Looper.java:205)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at android.app.ActivityThread.main(ActivityThread.java:6783)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at java.lang.reflect.Method.invoke(Native Method)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1225)
08-12 10:42:00.382 20271 20271 E unknown:ReactNative:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1086)
08-12 10:42:00.390 20271 20271 E AndroidRuntime: FATAL EXCEPTION: main
08-12 10:42:00.390 20271 20271 E AndroidRuntime: Process: com.pixeloapp, PID: 20271
```

**Comment by clamiax** [comment]
@jainkuniya Thanks for taking a look.

I'm still experiencing the issue and I also tried on a real device with the exact same effect (I made my first attempts on an emulator). Here is the relevant part my [logs](https://github.com/react-native-community/react-native-picker/files/3491546/adb.log). I don't know how to investigate further since I don't get any other useful error message.

**Comment by clamiax** [comment]
@jainkuniya I noticed that the line about picker was missing in my `package.json`. I also noticed that I was using react-native-picker instead of @react-native-community/picker. Not sure what the difference really is.

Anyway, after I ran the following commands the issue got solved:

    npm uninstall react-native-picker
    npm install --save @react-native-community/picker

Now everything is working properly. Thanks for your help and time.
